utils/providers.py

```diff
@@ -168,6 +171,15 @@ def setup_a_provider(prov_class=None, prov_type=None, validate=True, check_exist
         from cfme.infrastructure.provider import get_from_config
         providers = list_infra_providers()
 
+    final_providers = []
+    for provider in providers:
```

Comment:
Neither of these is the fault of this PR, but:
- the docs say set up a random provider. It'd be good to see some randomness in here,  probably by shuffling the provider list.
- problematic_providers doesn't need to be global'd